From 5b74d3e8005aae9df63bc207932da96380dbecf5 Mon Sep 17 00:00:00 2001
From: Linjie Fu <linjie.fu@intel.com>
Date: Wed, 8 Jan 2020 15:06:37 +0800
Subject: [PATCH 32/35] lavc/vaapi_hevc: add HEVCMain as default if profile is
 not found

Profile info may not be found for some reasons(such as profile_idc == 0).

Default to use VAProfileHEVCMain like ff_h265_guess_level did.

Signed-off-by: Linjie Fu <linjie.fu@intel.com>
---
 libavcodec/vaapi_hevc.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/libavcodec/vaapi_hevc.c b/libavcodec/vaapi_hevc.c
index 0604e23..ccd49e2 100644
--- a/libavcodec/vaapi_hevc.c
+++ b/libavcodec/vaapi_hevc.c
@@ -543,8 +543,12 @@ VAProfile ff_vaapi_parse_rext_profile(AVCodecContext *avctx)
     profile = ff_h265_get_profile(h265_raw_ptl);
     av_freep(&h265_raw_ptl);
 
-    if (!profile)
-        return VAProfileNone;
+    if (!profile) {
+        // Default to use Main profile.
+        av_log(avctx, AV_LOG_WARNING, "HEVC profile is not found, "
+               "default to use Main profile.\n");
+        return VAProfileHEVCMain;
+    }
 
 #if VA_CHECK_VERSION(1, 2, 0)
     if (!strcmp(profile->name, "Main 4:2:2 10") ||
-- 
2.7.4

